### PR TITLE
[SLM] UX `mlc_chat compile` improvement

### PR DESCRIPTION
**Pull request by davidpissarra**
This PR proposes changes to the new compilation CLI to complete A1 and A2 from #1364:

- Rename `max-window-size` to `context-window-size` (unless there is a reason, having different namings can be misleading since I believe both represent the same);
-  deprecate the usage of `config.json` in `mlc_chat compile`;
- use `mlc-chat-config.json` (created by `mlc_chat gen_mlc_chat_config`)  in `mlc_chat compile`;
- remove overlapped arguments from `mlc_chat compile` (namely `context-window-size`, `prefill_chunk_size`, `sliding_window` and `quantization`) and create `--overrides` argument to serve the purpose of overriding such settings in `mlc-chat-config.json` if needed.

I will add follow-up PRs to approach the next action items.

cc @junrushao @tqchen

### CODE REVIEW

**Review by tqchen — APPROVED** [review]

File: python/mlc_chat/support/config.py [hunk 1]
```diff
@@ -40,6 +40,17 @@ def from_dict(cls: Type[ConfigClass], source: Dict[str, Any]) -> ConfigClass:
         field_names = [field.name for field in dataclasses.fields(cls)]  # type: ignore[arg-type]
         fields = {k: v for k, v in source.items() if k in field_names}
         kwargs = {k: v for k, v in source.items() if k not in field_names}
+        if "model_config" in source:
+            fields |= {
```
[the remark this answers is not in the document]
Comment (tqchen) [reply python/mlc_chat/support/config.py:44]: good catch, @davidpissarra can you send a PR to update so we can be compatible before 3.10

[the remark this answers is not in the document]
Comment (davidpissarra) [reply python/mlc_chat/support/config.py:44]: Ah I see, I will note that so I can fix it in the next PR

**Comment by jinhongyii** [comment]
please update help message as well to reflect the API change of mlc_chat compile

**Comment by jinhongyii** [comment]
changing context window size to max window size breaks llm_chat.cc

**Pull request merged**